### PR TITLE
test: remove `cifar10_tf_keras` tests and examples.

### PR DESCRIPTION
**Pull request by ioga**
## Description

- Due to `cifar10_tf_keras` example no longer working well with modern tensorflow, remove it together with the tests.
- Move that example to the `determined-examples` repo: https://github.com/determined-ai/determined-examples/pull/3
- Update the `TFKerasTrial` docs to link to the `iris` example.
- Update the examples README and docs page to mention the new repo.

## Commentary (optional)

- this removes the only `TFKerasTrial` test we run as an end to end tests. there're still unit tests for `TFKerasTrial`.
- this also removes `test_tf_keras_parallel` which tested aggregation frequency and some profiling features.

are we comfortable with these removals? 

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by ioga** [comment]
(should be good for review now)

File: docs/get-started/example-solutions/_index.rst [hunk 1]
```diff
@@ -8,6 +8,9 @@ Get started quickly by using an example machine learning model that has been con
 Determined's APIs. Visit the ``examples/`` subdirectory of the `Determined GitHub repo
 <https://github.com/determined-ai/determined/tree/master/examples>`__ or download the link below.
```
Comment (MikhailKardash) [inline docs/get-started/example-solutions/_index.rst:9]: ```To view code examples maintained by the team, visit the ``examples/`` subdirectory of the `Determined GitHub repo
<https://github.com/determined-ai/determined/tree/master/examples>`__ or use one of the download links below```

File: examples/README.md [hunk 2]
```diff
@@ -1,18 +1,18 @@
 # Determined Examples
 
+Please find a large collection of community-sourced examples in
+[determined-ai/determined-examples](https://github.com/determined-ai/determined-examples/) repo.
+
+This directory only contains the examples which are routinely tested on CI.
```
Comment (MikhailKardash) [inline examples/README.md:6]: I'd swap the order here.

"This directory only contains examples that are routinely tested on our CI platform. Further examples are located in our community-sourced [determined-ai/determined-examples](https://github.com/determined-ai/determined-examples/) repo."

File: docs/get-started/example-solutions/_index.rst [hunk 3]
```diff
@@ -5,8 +5,13 @@
...
 Get started quickly by using an example machine learning model that has been converted to
-Determined's APIs. Visit the ``examples/`` subdirectory of the `Determined GitHub repo
-<https://github.com/determined-ai/determined/tree/master/examples>`__ or download the link below.
+Determined's APIs. To view code examples maintained by the team, visit the ``examples/``
+subdirectory of the `Determined GitHub repo
+<https://github.com/determined-ai/determined/tree/master/examples>`__ or use one of the download
+links below.
+
+For a large collection of community-contributed examples, visit the `determined-examples repo
+<https://github.com/determined-ai/determined-examples/>`__.
```
Comment (tara-det-ai) [inline docs/get-started/example-solutions/_index.rst:14]: ##########

Start with an example machine learning model converted to
Determined's APIs. Code examples are in the ``examples/``
subdirectory of the `Determined GitHub repo
<https://github.com/determined-ai/determined/tree/master/examples>`__. Download links are below.

For more examples, visit the `determined-examples repo
<https://github.com/determined-ai/determined-examples/>`__.

**Review by tara-det-ai — CHANGES_REQUESTED** [review]
suggested edits for simplification

File: docs/get-started/example-solutions/_index.rst (hunk 1 above)
Comment (tara-det-ai) [reply docs/get-started/example-solutions/_index.rst]: "team" is ambiguous

**Review by tara-det-ai — APPROVED** [review]
edits discussed

File: harness/tests/experiment/keras/test_tf_keras_trial.py [hunk 4]
```diff
@@ -368,82 +368,6 @@ def make_workloads() -> workload.Stream:
...
-        trainer = utils.TrainAndValidate()
-        yield from trainer.send(steps=2, validation_freq=2, scheduling_unit=1)
-
-        interceptor = workload.WorkloadResponseInterceptor()
-        yield from interceptor.send(workload.checkpoint_workload())
-        nonlocal latest_checkpoint, steps_completed
-        latest_checkpoint = interceptor.metrics_result()["uuid"]
-        steps_completed = trainer.get_steps_completed()
-
-    example_path = utils.cv_examples_path("cifar10_tf_keras/model_def.py")
```
Comment (MikhailKardash) [inline harness/tests/experiment/keras/test_tf_keras_trial.py:397]: Does swapping `iris` in-place not work here?

Comment (ioga) [reply harness/tests/experiment/keras/test_tf_keras_trial.py:397]: There seemed to be other unit tests which do exercise `TFKerasTrials`, so I thought there's no point in keeping this one if we have an e2e for it anyway. I can bring it back if you think it's valuable.

File: e2e_tests/tests/conftest.py [hunk 5]
```diff
@@ -52,7 +52,6 @@
     "e2e_gpu_quarantine",
     "model_hub_mmdetection_quarantine",
     "nightly_quarantine",
```
Comment (wes-turner) [inline e2e_tests/tests/conftest.py:54]: Here and in `pytest.ini`, also `nightly_quarantine`?

File: examples/README.md [hunk 6]
```diff
@@ -1,18 +1,19 @@
 # Determined Examples
 
+This directory only contains examples that are routinely tested on our CI platform.
```
Comment (wes-turner) [inline examples/README.md:3]: Nice. I like this addition.

File: harness/tests/experiment/keras/test_tf_keras_trial.py (hunk 4 above)
Comment (MikhailKardash) [reply harness/tests/experiment/keras/test_tf_keras_trial.py:397]: The `test-e2e-gpu-parallel` workflow only applies to main branch. To run this on dev branches, it has to be manually run in `test-e2e-longrunning`. I think keeping a unit test around for it is probably a good idea.

Comment (MikhailKardash) [reply harness/tests/experiment/keras/test_tf_keras_trial.py:397]: See: https://github.com/determined-ai/determined/blob/48874503af2b4dcc69b810721e1d5b7c2d34dfaa/.circleci/real_config.yml#L2742
and any other hits in that file containing `mark: ["parallel"]`

**Review by wes-turner — APPROVED** [review]
Tiny comment on something you may have missed, but other than that LGTM. Thanks for doing this.

File: harness/tests/requirements/requirements-harness.txt [hunk 7]
```diff
@@ -15,3 +15,4 @@ attrdict3
 moto
 # Pydantic V2 has changes that break existing tests
 pydantic<2
+pandas
```
Comment (MikhailKardash) [inline harness/tests/requirements/requirements-harness.txt:18]: Do we want this to be in `requirements-harness` or should this just be a pip install in `startup-hook.sh` ?

**Review by MikhailKardash — APPROVED** [review]
LGTM
